In [1]:
import pandas as pd
import numpy as np

In [2]:
impact_ad = pd.read_excel("Data/IMPACT_Study/ADSTART0_2023-05-25_06-46-11.xlsx")
impact_serum = pd.read_excel("Data/IMPACT_Study/Serum antibody_2023-05-25_06-32-15.xlsx")
impact_ige = pd.read_excel("Data/Impact_Study/IgE_IgG4_component_2023-05-25_06-30-40.xlsx")
impact_spt = pd.read_excel("Data/Impact_study/Skin Prick Test_2023-05-25_06-31-04.xlsx")

# About Data Sets

#### ADSTART0_2023-05-25_06-46-11 
- Overview of participant data 

#### IgE_IgG4_component_2023-05-25_06-30-40
- IgE component levels in this dataset  

#### Serum antibody_2023-05-25_06-32-15
OFC results:
- "Passed Visit 24 OFC for ITT"
- "Passed Visit 24 OFC No Imputation"	
- "Passed Visit 26 OFC for ITT"
- "Passed Visit 26 OFC No Imputation"

#### Skin Prick Test_2023-05-25_06-31-04
- "Wheal (mm)"

#### BAT data_2023-05-25_06-31-20
- Stands for "basophil activation test (BAT)"
- No known use case for this model 

---

# Acronyms 
- Intent-to-treat (ITT)
- Oral Immunotherapy (OIT)
- Initial Dose Escalation (IDE)

# About Study 
- taken from 2022 Lancet_IMPACT.pdf (pdf page 3)  

Children aged 12 months or older and younger than 
48 months were screened for inclusion in the study. 

__Inclusion criteria__ included the following: a clinical 
history of peanut allergy or avoidance without ever 
having eaten peanut, peanut-specific IgE levels of 5 kUA/L 
or higher, a skin prick test (SPT) wheal size greater than 
that of saline control by 3 mm or more, and a positive 
reaction to a cumulative dose of 500 mg or less of peanut 
in a double-blind, placebo-controlled food challenge 
(DBPCFC).   

__exclusion criteria__ included a history of 
severe anaphylaxis with hypotension to peanut, more 
than mild asthma or uncontrolled asthma, uncontrolled 
atopic dermatitis, and eosinophilic gastrointestinal 
disease (the full list of exclusion criteria is presented in 
the appendix p 2).

---
# Study Design

This is a randomized, double-blind, placebo-controlled, multi-center study comparing peanut oral immunotherapy to placebo. Eligible participants with peanut allergy will be randomly assigned to receive either peanut OIT or placebo for 134 weeks followed by peanut avoidance for 26 weeks.  

An initial blinded oral food challenge (OFC) to 1 g of peanut flour (500 mg peanut protein) will be conducted. Participants must have a clinical reaction during this blinded OFC to initiate study dosing. After the initial blinded OFC, the study design includes the following:  

__Initial Dose Escalation:__ This will occur on a single day in which multiple doses are given. Peanut or placebo dosing will be given incrementally and increase every 15-30 minutes until a dose of 12 mg peanut flour (6 mg peanut protein) or placebo flour is given. The first four doses will be administered as a peanut flour extract of 0.1 to 0.8 mg peanut protein, which is 10 to 80 microliters peanut flour extract, or placebo flour extract and the last three doses will be given as peanut flour of 3 to 12 mg peanut flour 1.5 to 6 mg peanut protein or placebo flour. Participants must tolerate a dose of at least 3 mg peanut flour (1.5 mg peanut protein) or placebo flour to remain in the study.  

__Build-up:__ After the initial dose escalation day, the participant will return to the research unit the next morning for an observed dose administration of the highest tolerated dose from the initial escalation day. The participant will then continue on the daily OIT dosing at home and return to the research unit every 2 weeks for a dose escalation. The dosing escalations will be consistent with previous similar OIT studies.
  
Participants who do not reach the 4000 mg peanut flour (2000 mg peanut protein) or placebo flour dose during the build-up phase may enter maintenance phase at their highest tolerated dose, which must be at least 500 mg peanut flour (250 mg peanut protein) or placebo flour.  

The build-up phase will comprise 30 weeks.  

__Maintenance:__ The participant will continue on daily OIT with return visits every 13 weeks. At the end of this phase the participant will undergo a blinded OFC to 10 g peanut flour (5 g peanut protein).  
This phase will comprise 104 weeks.  

__Avoidance:__ In this final phase participants stop OIT and will avoid peanut consumption They will be seen 2 weeks and 26 weeks after initiating this phase. At the completion of this phase participants will have a final blinded OFC to 10 g peanut flour (5 g peanut protein). Participants who do not have a clinical reaction to the challenge will receive an Open Food Challenge (OpFC).  

Avoidance will comprise 26 weeks.  

__Post-challenge:__ If participants do not have a clinical reaction during the OpFC at the end of avoidance, they will be allowed to consume peanut and will have one visit which will include peripheral blood sampling for mechanistic assays assessments.  

Post-challenge will comprise 2 weeks.  

---
# Exploring where and how OFCs are captured
Looking for 3 OFC tests in total

### 1st OFC: According to Study Design in protocol: 
-  Initial reaction: "An initial blinded oral food challenge (OFC) to 1 g of peanut flour (500 mg peanut protein) will be conducted. Participants must have a clinical reaction during this blinded OFC to initiate study dosing." (IDE)
- This was for 0.5 g peanut

### 2nd & 3rd OFC: According to Schedule of Assessments: Appendix 2:
- 5 g Oral Food Challenge performed during Avoidance phase during visit 24/week 134 and visit 26/week 160
- this was at or below 5 g peanut



# Filtering just the useful columns:

In [3]:
# impact_ad
# impact_serum 
# impact_ige 
# impact_spt

impact_ad_cols_to_keep = [
     'Participant ID',
     'Date of Screening Visit',
     'Randomized', # Use this to filter out participants that did not pass the study screening
     'Study Status', # Values: 'Discontinued Therapy', 'Completed Study', 'Enrolled but not Randomized', 'Early Termination', 'Screen Failure', 'Screened but not Enrolled'
     'Sex (character)',
     'Race',
     'Completed Study Assessments Numeric', #1 for yes/ 0 for no (means attended all visits up to 26, excluding 27)
     'Age at Screening (years) Not Rounded',
     'Study Termination Reason'
]

impact_serum_cols_to_keep = [
    'Participant ID',
    'Collection Date',
    'Visit',
    'Test Name', # Peanut IgE, Peanut IgE/Total IgE ratio, Peanut IgG4*, Peanut IgG4/IgE ratio, Total IgE
    'Unit',
    'Value', # results from 'Test Name'
]

impact_ige_cols_to_keep = [
    'Participant ID',
    'Collection Date',
    'Visit',
    'Antibody', #IgE, IgG4
    'Component', #rAra h 1, rAra h 2, rAra h 3, rAra h 6
    'Value',
    'Unit' 
]

impact_spt_cols_to_keep = [
    'Participant ID',
    'Date of Allergy Skin Test (Character)', # this is similar to 'Collection Date' in other datasets 
    'Visit',
    'Wheal (mm)',
    'OUT24NOI' # this will be used as OFC results for week 24
    #'OUT26NOI' # this will be used as OFC results for week 26
]

impact_ad = impact_ad[impact_ad_cols_to_keep]
impact_serum = impact_serum[impact_serum_cols_to_keep]
impact_ige = impact_ige[impact_ige_cols_to_keep]
impact_spt = impact_spt[impact_spt_cols_to_keep]


In [4]:
print(impact_ad.shape)
impact_ad.head()

(209, 9)


Participant ID Date of Screening Visit Randomized  \
0  IMPACT_101655              2012-10-11        Yes   
1  IMPACT_102436              2013-03-14        Yes   
2  IMPACT_105670              2013-04-04        Yes   
3  IMPACT_112496              2013-05-05         No   
4  IMPACT_113135              2013-11-25        Yes   

                  Study Status Sex (character)             Race  \
0         Discontinued Therapy            Male  White/Caucasian   
1              Completed Study            Male  White/Caucasian   
2              Completed Study          Female  White/Caucasian   
3  Enrolled but not Randomized          Female  White/Caucasian   
4            Early Termination          Female  White/Caucasian   

   Completed Study Assessments Numeric  Age at Screening (years) Not Rounded  \
0                                    0                                   3.8   
1                                    1                                   3.9   
2                                    1                                   3.0   
3                                    0                                   3.2   
4                                    0                                   2.6   

  Study Termination Reason  
0            Adverse Event  
1                      NaN  
2                      NaN  
3        Withdrawn Consent  
4        Withdrawn Consent

In [5]:
print(impact_serum.shape)
impact_serum.head()

(3069, 6)


Participant ID Collection Date  Visit                   Test Name    Unit  \
0  IMPACT_101655      2012-10-11     -2                  Peanut IgE    kU/L   
1  IMPACT_101655      2012-10-11     -2  Peanut IgE/Total IgE ratio   Ratio   
2  IMPACT_101655      2012-10-11     -2                Peanut IgG4*  mcg/mL   
3  IMPACT_101655      2012-10-11     -2       Peanut IgG4/IgE ratio   Ratio   
4  IMPACT_101655      2012-10-11     -2                   Total IgE   IU/mL   

       Value  
0  25.700000  
1  27.934783  
2   0.300000  
3   0.004864  
4  92.000000

In [6]:
print(impact_ige.shape)
impact_ige.head()

(4848, 7)


Participant ID Collection Date  Visit Antibody Component  Value   Unit
0  IMPACT_101655      2012-10-11     -2      IgE  rAra h 1   8.76   KU/L
1  IMPACT_101655      2012-10-11     -2      IgE  rAra h 2  27.50   KU/L
2  IMPACT_101655      2012-10-11     -2      IgE  rAra h 3   0.50   KU/L
3  IMPACT_101655      2012-10-11     -2      IgE  rAra h 6  17.30  kUA/L
4  IMPACT_101655      2012-10-11     -2     IgG4  rAra h 1   0.15   MG/L

In [7]:
print(impact_spt.shape)
impact_spt.head()

(621, 5)


Participant ID Date of Allergy Skin Test (Character)  Visit  Wheal (mm)  \
0  IMPACT_101655                            2012-10-11     -2        17.5   
1  IMPACT_101655                            2013-05-28     16         4.5   
2  IMPACT_102436                            2013-03-14     -2        16.0   
3  IMPACT_102436                            2013-11-26     16         9.5   
4  IMPACT_102436                            2014-11-28     20         NaN   

   OUT24NOI  
0       NaN  
1       NaN  
2       1.0  
3       1.0  
4       1.0

---
# Filtering participants that were study eligible (n=144)
- Total participants enrolled = 209  
- Total participants eligible = 144
- Total excluded = 65 (62 that did not have OFC fail results + 2 that could not handle the IDE)

#### filtering solution (this will help filter initial screening OFC results too)
- filter by'ADSTART0 -> immpact_ad['Randomized']=="Yes" 
- filter by ADSTART0 -> "Study Termination Reason"-> "Inability to reach 3 mg peanut/placebo flour (1.5 mg peanut/placebo protein) during the initial dose escalation" (These are the two that terminated after being randomized but could not reach their IDE: 146-2 -> n=144)

In [8]:
# filter out ineligible participants 
impact_ad_eligible = impact_ad.loc[(impact_ad['Randomized'] == 'Yes') & ~(impact_ad['Study Termination Reason'] == 'Inability to reach 3 mg peanut/placebo flour (1.5 mg peanut/placebo protein) during the initial dose escalation')]


In [9]:
# drop column
impact_ad_eligible = impact_ad_eligible.drop(columns=['Randomized'])

In [10]:
print(impact_ad_eligible.shape)
impact_ad_eligible.head()

(144, 8)


Participant ID Date of Screening Visit          Study Status  \
0  IMPACT_101655              2012-10-11  Discontinued Therapy   
1  IMPACT_102436              2013-03-14       Completed Study   
2  IMPACT_105670              2013-04-04       Completed Study   
4  IMPACT_113135              2013-11-25     Early Termination   
5  IMPACT_115876              2014-03-19       Completed Study   

  Sex (character)             Race  Completed Study Assessments Numeric  \
0            Male  White/Caucasian                                    0   
1            Male  White/Caucasian                                    1   
2          Female  White/Caucasian                                    1   
4          Female  White/Caucasian                                    0   
5            Male       Mixed Race                                    1   

   Age at Screening (years) Not Rounded Study Termination Reason  
0                                   3.8            Adverse Event  
1                                   3.9                      NaN  
2                                   3.0                      NaN  
4                                   2.6        Withdrawn Consent  
5                                   2.2                      NaN

# Obtain list of participants who were eligible 


In [11]:
eligible_participant_IDs = set(impact_ad_eligible['Participant ID'])

In [12]:
len(eligible_participant_IDs)

144

# Filtering each remaining dataset to include just the eligible participants 

In [13]:
impact_ige_eligible = impact_ige[impact_ige['Participant ID'].isin(eligible_participant_IDs)]
impact_serum_eligible = impact_serum[impact_serum['Participant ID'].isin(eligible_participant_IDs)]
impact_spt_eligible = impact_spt[impact_spt['Participant ID'].isin(eligible_participant_IDs)]

# all datasets so far with 144 eligible participants and relevant columns:
# impact_ad_eligible
# impact_ige_eligible 
# impact_serum_eligible
# impact_spt_eligible

In [14]:
print("impact_ad_eligible")
print(impact_ad_eligible.shape)

print("impact_ige_eligible")
print(impact_ige_eligible.shape)

print("impact_serum_eligible")
print(impact_serum_eligible.shape)

print("impact_spt_eligible")
print(impact_spt_eligible.shape)

print("impact_ad_eligible")
print(len(impact_ad_eligible['Participant ID'].unique().tolist()))

print("impact_ige_eligible")
print(len(impact_ige_eligible['Participant ID'].unique().tolist()))

print("impact_serum_eligible")
print(len(impact_serum_eligible['Participant ID'].unique().tolist()))

print("impact_spt_eligible")
print(len(impact_spt_eligible['Participant ID'].unique().tolist()))

impact_ad_eligible
(144, 8)
impact_ige_eligible
(4832, 7)
impact_serum_eligible
(3059, 6)
impact_spt_eligible
(619, 5)
impact_ad_eligible
144
impact_ige_eligible
144
impact_serum_eligible
144
impact_spt_eligible
144


In [15]:
print(impact_ad_eligible.shape)
impact_ad_eligible.head()

(144, 8)


Participant ID Date of Screening Visit          Study Status  \
0  IMPACT_101655              2012-10-11  Discontinued Therapy   
1  IMPACT_102436              2013-03-14       Completed Study   
2  IMPACT_105670              2013-04-04       Completed Study   
4  IMPACT_113135              2013-11-25     Early Termination   
5  IMPACT_115876              2014-03-19       Completed Study   

  Sex (character)             Race  Completed Study Assessments Numeric  \
0            Male  White/Caucasian                                    0   
1            Male  White/Caucasian                                    1   
2          Female  White/Caucasian                                    1   
4          Female  White/Caucasian                                    0   
5            Male       Mixed Race                                    1   

   Age at Screening (years) Not Rounded Study Termination Reason  
0                                   3.8            Adverse Event  
1                                   3.9                      NaN  
2                                   3.0                      NaN  
4                                   2.6        Withdrawn Consent  
5                                   2.2                      NaN

# Standardizing Column Names Across Datasets



In [16]:
# impact_ad_eligible
# impact_ige_eligible 
# impact_serum_eligible
# impact_spt_eligible

# renaming'Date of Screening Visit' to 'Collection Date' to match the other datasets 
# impact_ad_eligible = impact_ad_eligible.rename(columns={'Date of Screening Visit': 'Collection Date'})
impact_spt_eligible = impact_spt_eligible.rename(columns={'Date of Allergy Skin Test (Character)': 'Collection Date'})


In [17]:
print(impact_spt_eligible.shape)
impact_spt_eligible.head()

(619, 5)


Participant ID Collection Date  Visit  Wheal (mm)  OUT24NOI
0  IMPACT_101655      2012-10-11     -2        17.5       NaN
1  IMPACT_101655      2013-05-28     16         4.5       NaN
2  IMPACT_102436      2013-03-14     -2        16.0       1.0
3  IMPACT_102436      2013-11-26     16         9.5       1.0
4  IMPACT_102436      2014-11-28     20         NaN       1.0

# Filtering all participants except those who made it to week 24

- per IMPACT chart, 114 participants made it to Visit 24

filtering results below shows 116 participants common to all datasets. Will remove the two that did not have results manually 


In [18]:
# impact_ige_eligible 

In [19]:
impact_ige_eligible_24 = impact_ige_eligible[impact_ige_eligible['Visit'] == 24]

In [20]:
print(impact_ige_eligible_24.shape)
print(len(impact_ige_eligible_24['Participant ID'].unique().tolist()))
impact_ige_eligible_24.head()

(936, 7)
117


Participant ID Collection Date  Visit Antibody Component  Value   Unit
40  IMPACT_102436      2015-11-20     24      IgE  rAra h 1   1.14  kUA/L
41  IMPACT_102436      2015-11-20     24      IgE  rAra h 2   8.67  kUA/L
42  IMPACT_102436      2015-11-20     24      IgE  rAra h 3   0.05  kUA/L
43  IMPACT_102436      2015-11-20     24      IgE  rAra h 6   6.64  kUA/L
44  IMPACT_102436      2015-11-20     24     IgG4  rAra h 1  12.90   MG/L

In [21]:
# impact_serum_eligible

In [22]:
impact_serum_eligible_24 = impact_serum_eligible[impact_serum_eligible['Visit'] == 24]

In [23]:
print(impact_serum_eligible_24.shape)
print(len(impact_serum_eligible_24['Participant ID'].unique().tolist()))
impact_serum_eligible_24.head()

(590, 6)
118


Participant ID Collection Date  Visit                   Test Name    Unit  \
25  IMPACT_102436      2015-11-20     24                  Peanut IgE    kU/L   
26  IMPACT_102436      2015-11-20     24  Peanut IgE/Total IgE ratio   Ratio   
27  IMPACT_102436      2015-11-20     24                Peanut IgG4*  mcg/mL   
28  IMPACT_102436      2015-11-20     24       Peanut IgG4/IgE ratio   Ratio   
29  IMPACT_102436      2015-11-20     24                   Total IgE   IU/mL   

        Value  
25   8.410000  
26   8.670103  
27  90.200000  
28   4.468886  
29  97.000000

In [24]:
# impact_spt_eligible

In [25]:
impact_spt_eligible_24 = impact_spt_eligible[impact_spt_eligible['Visit'] == 24]

In [26]:
print(impact_spt_eligible_24.shape)
print(len(impact_spt_eligible_24['Participant ID'].unique().tolist()))
impact_spt_eligible_24.head()

(118, 5)
118


Participant ID Collection Date  Visit  Wheal (mm)  OUT24NOI
5   IMPACT_102436      2015-11-20     24         2.5       1.0
10  IMPACT_105670      2015-11-25     24         0.0       1.0
15  IMPACT_113135      2016-09-12     24        24.5       0.0
20  IMPACT_115876      2016-11-09     24        27.5       0.0
25  IMPACT_136775      2017-11-20     24         7.5       0.0

In [27]:
# finding common participants who reached visit 24 

# impact_ige_eligible_24
# impact_serum_eligible_24
# impact_spt_eligible_24

ige_list_24 = impact_ige_eligible_24['Participant ID'].tolist()
serum_list_24 = impact_serum_eligible_24['Participant ID'].tolist()
spt_list_24 = impact_spt_eligible_24['Participant ID'].tolist()

common_participants_24 = set(spt_list_24).intersection(ige_list_24, serum_list_24)
common_participants_24 = list(common_participants_24)



In [28]:
len(common_participants_24)
#output 141

117

In [29]:

# Updating each data frame so that they only contain the common participants 
impact_ad_eligible_24_common = impact_ad_eligible[impact_ad_eligible['Participant ID'].isin(common_participants_24)]
impact_ige_eligible_24_common = impact_ige_eligible_24[impact_ige_eligible_24['Participant ID'].isin(common_participants_24)]
impact_serum_eligible_24_common = impact_serum_eligible_24[impact_serum_eligible_24['Participant ID'].isin(common_participants_24)]
impact_spt_eligible_24_common = impact_spt_eligible_24[impact_spt_eligible_24['Participant ID'].isin(common_participants_24)]



In [30]:
print(impact_ad_eligible_24_common.shape)
print(len(impact_ad_eligible_24_common['Participant ID'].unique().tolist()))
impact_ad_eligible_24_common.head()

(117, 8)
117


Participant ID Date of Screening Visit       Study Status Sex (character)  \
1  IMPACT_102436              2013-03-14    Completed Study            Male   
2  IMPACT_105670              2013-04-04    Completed Study          Female   
4  IMPACT_113135              2013-11-25  Early Termination          Female   
5  IMPACT_115876              2014-03-19    Completed Study            Male   
9  IMPACT_136775              2015-03-28  Early Termination          Female   

              Race  Completed Study Assessments Numeric  \
1  White/Caucasian                                    1   
2  White/Caucasian                                    1   
4  White/Caucasian                                    0   
5       Mixed Race                                    1   
9       Mixed Race                                    0   

   Age at Screening (years) Not Rounded Study Termination Reason  
1                                   3.9                      NaN  
2                                   3.0                      NaN  
4                                   2.6        Withdrawn Consent  
5                                   2.2                      NaN  
9                                   3.5        Withdrawn Consent

In [31]:
print(impact_ige_eligible_24_common.shape)
print(len(impact_ige_eligible_24_common['Participant ID'].unique().tolist()))
impact_ige_eligible_24_common.head()

(936, 7)
117


Participant ID Collection Date  Visit Antibody Component  Value   Unit
40  IMPACT_102436      2015-11-20     24      IgE  rAra h 1   1.14  kUA/L
41  IMPACT_102436      2015-11-20     24      IgE  rAra h 2   8.67  kUA/L
42  IMPACT_102436      2015-11-20     24      IgE  rAra h 3   0.05  kUA/L
43  IMPACT_102436      2015-11-20     24      IgE  rAra h 6   6.64  kUA/L
44  IMPACT_102436      2015-11-20     24     IgG4  rAra h 1  12.90   MG/L

In [32]:
print(impact_serum_eligible_24_common.shape)
print(len(impact_serum_eligible_24_common['Participant ID'].unique().tolist()))
impact_serum_eligible_24_common.head()

(585, 6)
117


Participant ID Collection Date  Visit                   Test Name    Unit  \
25  IMPACT_102436      2015-11-20     24                  Peanut IgE    kU/L   
26  IMPACT_102436      2015-11-20     24  Peanut IgE/Total IgE ratio   Ratio   
27  IMPACT_102436      2015-11-20     24                Peanut IgG4*  mcg/mL   
28  IMPACT_102436      2015-11-20     24       Peanut IgG4/IgE ratio   Ratio   
29  IMPACT_102436      2015-11-20     24                   Total IgE   IU/mL   

        Value  
25   8.410000  
26   8.670103  
27  90.200000  
28   4.468886  
29  97.000000

In [33]:
print(impact_spt_eligible_24_common.shape)
print(len(impact_spt_eligible_24_common['Participant ID'].unique().tolist()))
impact_spt_eligible_24_common.head()

(117, 5)
117


Participant ID Collection Date  Visit  Wheal (mm)  OUT24NOI
5   IMPACT_102436      2015-11-20     24         2.5       1.0
10  IMPACT_105670      2015-11-25     24         0.0       1.0
15  IMPACT_113135      2016-09-12     24        24.5       0.0
20  IMPACT_115876      2016-11-09     24        27.5       0.0
25  IMPACT_136775      2017-11-20     24         7.5       0.0

# Calculating Age at visit 24



In [34]:
# transforming the "Age at Screening (years) Not Rounded" column in "impact_ad_eligible_24_common" to be in months


#convering the 'Age at Screening (years) Not Rounded' to months 
impact_ad_eligible_24_common['Age at Screening (years) Not Rounded'] = impact_ad_eligible_24_common['Age at Screening (years) Not Rounded'] * 12

print(impact_ad_eligible_24_common.shape)
impact_ad_eligible_24_common.head()


(117, 8)


<ipython-input-34-f53c2c19419b>:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  impact_ad_eligible_24_common['Age at Screening (years) Not Rounded'] = impact_ad_eligible_24_common['Age at Screening (years) Not Rounded'] * 12


Participant ID Date of Screening Visit       Study Status Sex (character)  \
1  IMPACT_102436              2013-03-14    Completed Study            Male   
2  IMPACT_105670              2013-04-04    Completed Study          Female   
4  IMPACT_113135              2013-11-25  Early Termination          Female   
5  IMPACT_115876              2014-03-19    Completed Study            Male   
9  IMPACT_136775              2015-03-28  Early Termination          Female   

              Race  Completed Study Assessments Numeric  \
1  White/Caucasian                                    1   
2  White/Caucasian                                    1   
4  White/Caucasian                                    0   
5       Mixed Race                                    1   
9       Mixed Race                                    0   

   Age at Screening (years) Not Rounded Study Termination Reason  
1                                  46.8                      NaN  
2                                  36.0                      NaN  
4                                  31.2        Withdrawn Consent  
5                                  26.4                      NaN  
9                                  42.0        Withdrawn Consent

In [35]:
# Using OFC Results from SPT to Calculate date difference 

In [36]:
impact_spt_eligible_24_common.head()

Participant ID Collection Date  Visit  Wheal (mm)  OUT24NOI
5   IMPACT_102436      2015-11-20     24         2.5       1.0
10  IMPACT_105670      2015-11-25     24         0.0       1.0
15  IMPACT_113135      2016-09-12     24        24.5       0.0
20  IMPACT_115876      2016-11-09     24        27.5       0.0
25  IMPACT_136775      2017-11-20     24         7.5       0.0

In [37]:
#Merging ad and spt  
merged_ad_spt = impact_ad_eligible_24_common.merge(impact_spt_eligible_24_common, 
                                                               how='outer', #preserves all rows and adds NaN for empty rows
                                                               on=['Participant ID'])


In [38]:
# calculate the time difference in months and add it to 'Age at Screening (years) Not Rounded'
merged_ad_spt['Age at OFC 24'] = merged_ad_spt.apply(
    lambda row: row['Age at Screening (years) Not Rounded'] + pd.Timedelta(row['Collection Date'] - row['Date of Screening Visit']).days / 30, axis=1
)


In [39]:
# round the 'Age at OFC 24' column to the nearest month
merged_ad_spt['Age at OFC 24'] = merged_ad_spt['Age at OFC 24'].round(1)



In [40]:
# dropping initial screening column (and other unecessary cols)

merged_ad_spt = merged_ad_spt.drop(columns = ['Date of Screening Visit', 
                                              'Completed Study Assessments Numeric', 
                                              'Study Termination Reason', 
                                              'Study Status',
                                              'Age at Screening (years) Not Rounded'
                                             ])

In [41]:
print(merged_ad_spt.shape)
merged_ad_spt.head()

(117, 8)


Participant ID Sex (character)             Race Collection Date  Visit  \
0  IMPACT_102436            Male  White/Caucasian      2015-11-20     24   
1  IMPACT_105670          Female  White/Caucasian      2015-11-25     24   
2  IMPACT_113135          Female  White/Caucasian      2016-09-12     24   
3  IMPACT_115876            Male       Mixed Race      2016-11-09     24   
4  IMPACT_136775          Female       Mixed Race      2017-11-20     24   

   Wheal (mm)  OUT24NOI  Age at OFC 24  
0         2.5       1.0           79.5  
1         0.0       1.0           68.2  
2        24.5       0.0           65.3  
3        27.5       0.0           58.6  
4         7.5       0.0           74.3

# 3 datasets left to merge


In [42]:
# 3 datasets left to merge: 
# merged_ad_spt -> contains AGE at 24
# impact_ige_eligible_24_common -> contains components 
# impact_serum_eligible_24_common -> contains Peanut & Total IgE 



In [43]:
print(merged_ad_spt.shape)
merged_ad_spt.head()

(117, 8)


Participant ID Sex (character)             Race Collection Date  Visit  \
0  IMPACT_102436            Male  White/Caucasian      2015-11-20     24   
1  IMPACT_105670          Female  White/Caucasian      2015-11-25     24   
2  IMPACT_113135          Female  White/Caucasian      2016-09-12     24   
3  IMPACT_115876            Male       Mixed Race      2016-11-09     24   
4  IMPACT_136775          Female       Mixed Race      2017-11-20     24   

   Wheal (mm)  OUT24NOI  Age at OFC 24  
0         2.5       1.0           79.5  
1         0.0       1.0           68.2  
2        24.5       0.0           65.3  
3        27.5       0.0           58.6  
4         7.5       0.0           74.3

In [44]:
print(impact_ige_eligible_24_common.shape)
impact_ige_eligible_24_common.head()

(936, 7)


Participant ID Collection Date  Visit Antibody Component  Value   Unit
40  IMPACT_102436      2015-11-20     24      IgE  rAra h 1   1.14  kUA/L
41  IMPACT_102436      2015-11-20     24      IgE  rAra h 2   8.67  kUA/L
42  IMPACT_102436      2015-11-20     24      IgE  rAra h 3   0.05  kUA/L
43  IMPACT_102436      2015-11-20     24      IgE  rAra h 6   6.64  kUA/L
44  IMPACT_102436      2015-11-20     24     IgG4  rAra h 1  12.90   MG/L

In [45]:
print(impact_serum_eligible_24_common.shape)
impact_serum_eligible_24_common.head()

(585, 6)


Participant ID Collection Date  Visit                   Test Name    Unit  \
25  IMPACT_102436      2015-11-20     24                  Peanut IgE    kU/L   
26  IMPACT_102436      2015-11-20     24  Peanut IgE/Total IgE ratio   Ratio   
27  IMPACT_102436      2015-11-20     24                Peanut IgG4*  mcg/mL   
28  IMPACT_102436      2015-11-20     24       Peanut IgG4/IgE ratio   Ratio   
29  IMPACT_102436      2015-11-20     24                   Total IgE   IU/mL   

        Value  
25   8.410000  
26   8.670103  
27  90.200000  
28   4.468886  
29  97.000000

# Breaking out components into columns 
in "impact_ige_eligible_24_common"

In [46]:
# dropping IgG4 from Antibody column, only want IgE
impact_ige_eligible_24_common = impact_ige_eligible_24_common[impact_ige_eligible_24_common['Antibody']=='IgE']


In [47]:
# Creating new columns for each component

# Create new columns with initial NaN values
impact_ige_eligible_24_common['Ara h1 (kU/L)'] = np.nan
impact_ige_eligible_24_common['Ara h2 (kU/L)'] = np.nan
impact_ige_eligible_24_common['Ara h3 (kU/L)'] = np.nan
impact_ige_eligible_24_common['Ara h6 (kU/L)'] = np.nan

# Populate the new columns based on conditions
impact_ige_eligible_24_common.loc[impact_ige_eligible_24_common['Component'] == 'rAra h 1', 'Ara h1 (kU/L)'] = impact_ige_eligible_24_common.loc[impact_ige_eligible_24_common['Component'] == 'rAra h 1', 'Value']
impact_ige_eligible_24_common.loc[impact_ige_eligible_24_common['Component'] == 'rAra h 2', 'Ara h2 (kU/L)'] = impact_ige_eligible_24_common.loc[impact_ige_eligible_24_common['Component'] == 'rAra h 2', 'Value']
impact_ige_eligible_24_common.loc[impact_ige_eligible_24_common['Component'] == 'rAra h 3', 'Ara h3 (kU/L)'] = impact_ige_eligible_24_common.loc[impact_ige_eligible_24_common['Component'] == 'rAra h 3', 'Value']
impact_ige_eligible_24_common.loc[impact_ige_eligible_24_common['Component'] == 'rAra h 6', 'Ara h6 (kU/L)'] = impact_ige_eligible_24_common.loc[impact_ige_eligible_24_common['Component'] == 'rAra h 6', 'Value']

# Drop the specified columns
impact_ige_eligible_24_common = impact_ige_eligible_24_common.drop(columns=['Antibody', 'Component', 'Value', 'Unit'])



In [48]:
print(impact_ige_eligible_24_common.shape)
impact_ige_eligible_24_common.head(10)


(468, 7)


Participant ID Collection Date  Visit  Ara h1 (kU/L)  Ara h2 (kU/L)  \
40   IMPACT_102436      2015-11-20     24           1.14            NaN   
41   IMPACT_102436      2015-11-20     24            NaN           8.67   
42   IMPACT_102436      2015-11-20     24            NaN            NaN   
43   IMPACT_102436      2015-11-20     24            NaN            NaN   
72   IMPACT_105670      2015-11-25     24           1.95            NaN   
73   IMPACT_105670      2015-11-25     24            NaN          10.40   
74   IMPACT_105670      2015-11-25     24            NaN            NaN   
75   IMPACT_105670      2015-11-25     24            NaN            NaN   
112  IMPACT_113135      2016-09-12     24          16.30            NaN   
113  IMPACT_113135      2016-09-12     24            NaN         181.00   

     Ara h3 (kU/L)  Ara h6 (kU/L)  
40             NaN            NaN  
41             NaN            NaN  
42            0.05            NaN  
43             NaN           6.64  
72             NaN            NaN  
73             NaN            NaN  
74            0.39            NaN  
75             NaN           4.97  
112            NaN            NaN  
113            NaN            NaN

In [49]:
# Merging duplicate rows and overwriting NaN values

# Group by columns and aggregate using the mean (for numeric columns) or first (for non-numeric columns)
impact_ige_eligible_24_common = impact_ige_eligible_24_common.groupby(['Participant ID', 'Collection Date', 'Visit'], as_index=False).agg({'Ara h1 (kU/L)': 'mean', 'Ara h2 (kU/L)': 'mean', 'Ara h3 (kU/L)': 'mean', 'Ara h6 (kU/L)': 'mean'})


# Reset the index
impact_ige_eligible_24_common.reset_index(drop=True, inplace=True)



In [50]:
print(impact_ige_eligible_24_common.shape)
impact_ige_eligible_24_common.head(20)


(117, 7)


Participant ID Collection Date  Visit  Ara h1 (kU/L)  Ara h2 (kU/L)  \
0   IMPACT_102436      2015-11-20     24           1.14           8.67   
1   IMPACT_105670      2015-11-25     24           1.95          10.40   
2   IMPACT_113135      2016-09-12     24          16.30         181.00   
3   IMPACT_115876      2016-11-09     24           8.75           9.54   
4   IMPACT_136775      2017-11-20     24           0.88           7.80   
5   IMPACT_139237      2017-08-23     24          19.20         102.00   
6   IMPACT_149018      2017-04-17     24          13.90          15.30   
7   IMPACT_196748      2017-01-29     24           0.05           0.52   
8   IMPACT_205488      2015-08-08     24           0.20           1.26   
9   IMPACT_207729      2017-06-13     24          13.10           6.72   
10  IMPACT_226208      2015-10-06     24           0.28          10.60   
11  IMPACT_228754      2017-05-21     24           4.34           6.52   
12  IMPACT_255737      2015-08-27     24         308.00         258.00   
13  IMPACT_256280      2016-09-11     24           3.76          12.60   
14  IMPACT_266884      2017-11-06     24          12.70          47.20   
15  IMPACT_272641      2016-08-24     24           0.14           3.48   
16  IMPACT_283272      2017-01-14     24          43.30          60.90   
17  IMPACT_289709      2016-06-05     24           6.92          11.80   
18  IMPACT_291362      2016-07-26     24         163.00         307.00   
19  IMPACT_298007      2016-07-23     24           0.28           5.48   

    Ara h3 (kU/L)  Ara h6 (kU/L)  
0            0.05           6.64  
1            0.39           4.97  
2            1.40         190.00  
3            0.05          11.00  
4            0.05           4.93  
5            2.28          62.20  
6            1.06          12.30  
7            0.05           0.66  
8            0.05           3.00  
9            0.05          11.30  
10           0.17           7.35  
11           0.22           6.95  
12          27.80         179.00  
13           1.61          18.20  
14           2.20          34.30  
15           0.32           4.82  
16           0.35          49.20  
17           1.41           8.96  
18          56.50         153.00  
19           0.05           3.58

In [51]:

print(len(impact_ige_eligible_24_common['Participant ID'].unique().tolist()))



117


# Merging "impact_ige_eligible_24_common" with "merged_ad_spt"
- both are 117 rows 

## Result -> One participant needs to be manually merged:
(23)  -   IMPACT_320824  
(117) -    IMPACT_320824

In [52]:
merged_ad_spt.head()

Participant ID Sex (character)             Race Collection Date  Visit  \
0  IMPACT_102436            Male  White/Caucasian      2015-11-20     24   
1  IMPACT_105670          Female  White/Caucasian      2015-11-25     24   
2  IMPACT_113135          Female  White/Caucasian      2016-09-12     24   
3  IMPACT_115876            Male       Mixed Race      2016-11-09     24   
4  IMPACT_136775          Female       Mixed Race      2017-11-20     24   

   Wheal (mm)  OUT24NOI  Age at OFC 24  
0         2.5       1.0           79.5  
1         0.0       1.0           68.2  
2        24.5       0.0           65.3  
3        27.5       0.0           58.6  
4         7.5       0.0           74.3

In [53]:
#Merging ad and spt  
merged_ad_spt_ige = merged_ad_spt.merge(impact_ige_eligible_24_common, 
                                                               how='outer', #preserves all rows and adds NaN for empty rows
                                                               on=['Participant ID', 'Visit', 'Collection Date'])


In [54]:
print(merged_ad_spt_ige.shape)
merged_ad_spt_ige.head()

(118, 12)


Participant ID Sex (character)             Race Collection Date  Visit  \
0  IMPACT_102436            Male  White/Caucasian      2015-11-20     24   
1  IMPACT_105670          Female  White/Caucasian      2015-11-25     24   
2  IMPACT_113135          Female  White/Caucasian      2016-09-12     24   
3  IMPACT_115876            Male       Mixed Race      2016-11-09     24   
4  IMPACT_136775          Female       Mixed Race      2017-11-20     24   

   Wheal (mm)  OUT24NOI  Age at OFC 24  Ara h1 (kU/L)  Ara h2 (kU/L)  \
0         2.5       1.0           79.5           1.14           8.67   
1         0.0       1.0           68.2           1.95          10.40   
2        24.5       0.0           65.3          16.30         181.00   
3        27.5       0.0           58.6           8.75           9.54   
4         7.5       0.0           74.3           0.88           7.80   

   Ara h3 (kU/L)  Ara h6 (kU/L)  
0           0.05           6.64  
1           0.39           4.97  
2           1.40         190.00  
3           0.05          11.00  
4           0.05           4.93

In [55]:
#seeing which row is duplicated

duplicated_ids = merged_ad_spt_ige['Participant ID'].duplicated(keep=False)
duplicated_values = merged_ad_spt_ige[duplicated_ids]

print(duplicated_values['Participant ID'])


23     IMPACT_320824
117    IMPACT_320824
Name: Participant ID, dtype: object


# Breaking out Peanut IgE/Total IgE
- in "impact_serum_eligible_24_common"


In [56]:
# removing rows from 'Test Name' for everything except 'Peanut IgE' and 'Total IgE'

#defining list to keep
keep_IgEs = ['Peanut IgE', 'Total IgE']

impact_serum_eligible_24_common = impact_serum_eligible_24_common[impact_serum_eligible_24_common['Test Name'].isin(keep_IgEs)]



In [57]:

impact_serum_eligible_24_common['Test Name'].unique()


array(['Peanut IgE', 'Total IgE'], dtype=object)

In [58]:


print(impact_serum_eligible_24_common.shape)
impact_serum_eligible_24_common.head()

(234, 6)


Participant ID Collection Date  Visit   Test Name   Unit   Value
25  IMPACT_102436      2015-11-20     24  Peanut IgE   kU/L    8.41
29  IMPACT_102436      2015-11-20     24   Total IgE  IU/mL   97.00
50  IMPACT_105670      2015-11-25     24  Peanut IgE   kU/L   10.70
54  IMPACT_105670      2015-11-25     24   Total IgE  IU/mL   70.00
75  IMPACT_113135      2016-09-12     24  Peanut IgE   kU/L  243.00

In [59]:
# Create new columns with initial NaN values
impact_serum_eligible_24_common['Peanut IgE (kU/L)'] = np.nan
impact_serum_eligible_24_common['Total IgE (kU/L)'] = np.nan

# Populate the new columns based on conditions
impact_serum_eligible_24_common.loc[impact_serum_eligible_24_common['Test Name'] == 'Peanut IgE', 'Peanut IgE (kU/L)'] = impact_serum_eligible_24_common.loc[impact_serum_eligible_24_common['Test Name'] == 'Peanut IgE', 'Value']
impact_serum_eligible_24_common.loc[impact_serum_eligible_24_common['Test Name'] == 'Total IgE', 'Total IgE (kU/L)'] = impact_serum_eligible_24_common.loc[impact_serum_eligible_24_common['Test Name'] == 'Total IgE', 'Value']

# Drop the specified columns
impact_serum_eligible_24_common = impact_serum_eligible_24_common.drop(columns=['Test Name', 'Unit', 'Value'])



In [60]:

len(impact_serum_eligible_24_common['Participant ID'].unique().tolist())


117

In [61]:
print(impact_serum_eligible_24_common.shape) # (234, 5)
impact_serum_eligible_24_common.head()
# note if all of the 144 participants had 2 rows for Peanut and Total, we'd have 232, but we have 234


(234, 5)


Participant ID Collection Date  Visit  Peanut IgE (kU/L)  Total IgE (kU/L)
25  IMPACT_102436      2015-11-20     24               8.41               NaN
29  IMPACT_102436      2015-11-20     24                NaN              97.0
50  IMPACT_105670      2015-11-25     24              10.70               NaN
54  IMPACT_105670      2015-11-25     24                NaN              70.0
75  IMPACT_113135      2016-09-12     24             243.00               NaN

In [63]:
# Merging duplicate rows and overwriting NaN values

# Group by columns and aggregate using the mean (for numeric columns) or first (for non-numeric columns)
impact_serum_eligible_24_common = impact_serum_eligible_24_common.groupby(['Participant ID', 'Collection Date', 'Visit'], as_index=False).agg({'Peanut IgE (kU/L)': 'mean', 'Total IgE (kU/L)': 'mean'})

# Reset the index
impact_serum_eligible_24_common.reset_index(drop=True, inplace=True)


In [66]:
print(impact_serum_eligible_24_common.shape) # correctly removed duplicates -> 117 rows  remain 
print(len(impact_serum_eligible_24_common['Participant ID'].unique().tolist()))
impact_serum_eligible_24_common.head(10)


(117, 5)
117


Participant ID Collection Date  Visit  Peanut IgE (kU/L)  Total IgE (kU/L)
0  IMPACT_102436      2015-11-20     24               8.41              97.0
1  IMPACT_105670      2015-11-25     24              10.70              70.0
2  IMPACT_113135      2016-09-12     24             243.00             565.0
3  IMPACT_115876      2016-11-09     24              19.20             727.0
4  IMPACT_136775      2017-11-20     24               9.06              25.0
5  IMPACT_139237      2017-08-23     24             129.00             399.0
6  IMPACT_149018      2017-04-17     24              26.80             542.0
7  IMPACT_196748      2017-01-29     24               0.73              80.0
8  IMPACT_205488      2015-08-08     24               3.61            1211.0
9  IMPACT_207729      2017-06-13     24              23.20             896.0

# Merging "merged_ad_spt_ige" with "impact_serum_eligible_24_common"

- both have 117 rows

In [67]:
merged_ad_spt_ige.head()

Participant ID Sex (character)             Race Collection Date  Visit  \
0  IMPACT_102436            Male  White/Caucasian      2015-11-20     24   
1  IMPACT_105670          Female  White/Caucasian      2015-11-25     24   
2  IMPACT_113135          Female  White/Caucasian      2016-09-12     24   
3  IMPACT_115876            Male       Mixed Race      2016-11-09     24   
4  IMPACT_136775          Female       Mixed Race      2017-11-20     24   

   Wheal (mm)  OUT24NOI  Age at OFC 24  Ara h1 (kU/L)  Ara h2 (kU/L)  \
0         2.5       1.0           79.5           1.14           8.67   
1         0.0       1.0           68.2           1.95          10.40   
2        24.5       0.0           65.3          16.30         181.00   
3        27.5       0.0           58.6           8.75           9.54   
4         7.5       0.0           74.3           0.88           7.80   

   Ara h3 (kU/L)  Ara h6 (kU/L)  
0           0.05           6.64  
1           0.39           4.97  
2           1.40         190.00  
3           0.05          11.00  
4           0.05           4.93

In [69]:
#Merging ad and spt  
merged_ad_spt_ige_serum = merged_ad_spt_ige.merge(impact_serum_eligible_24_common, 
                                                               how='outer', #preserves all rows and adds NaN for empty rows
                                                               on=['Participant ID', 'Visit', 'Collection Date'])


In [70]:
print(merged_ad_spt_ige_serum.shape)
print(len(merged_ad_spt_ige_serum['Participant ID'].unique().tolist()))
merged_ad_spt_ige_serum.head()

(118, 14)
117


Participant ID Sex (character)             Race Collection Date  Visit  \
0  IMPACT_102436            Male  White/Caucasian      2015-11-20     24   
1  IMPACT_105670          Female  White/Caucasian      2015-11-25     24   
2  IMPACT_113135          Female  White/Caucasian      2016-09-12     24   
3  IMPACT_115876            Male       Mixed Race      2016-11-09     24   
4  IMPACT_136775          Female       Mixed Race      2017-11-20     24   

   Wheal (mm)  OUT24NOI  Age at OFC 24  Ara h1 (kU/L)  Ara h2 (kU/L)  \
0         2.5       1.0           79.5           1.14           8.67   
1         0.0       1.0           68.2           1.95          10.40   
2        24.5       0.0           65.3          16.30         181.00   
3        27.5       0.0           58.6           8.75           9.54   
4         7.5       0.0           74.3           0.88           7.80   

   Ara h3 (kU/L)  Ara h6 (kU/L)  Peanut IgE (kU/L)  Total IgE (kU/L)  
0           0.05           6.64               8.41              97.0  
1           0.39           4.97              10.70              70.0  
2           1.40         190.00             243.00             565.0  
3           0.05          11.00              19.20             727.0  
4           0.05           4.93               9.06              25.0

# Final transformation of 'merged_ad_spt_ige_serum' to match LEAP columns 

In [80]:
impact_all_24 = merged_ad_spt_ige_serum

In [81]:
print(impact_all_24.shape)
impact_all_24.head()

(118, 14)


Participant ID Sex (character)             Race Collection Date  Visit  \
0  IMPACT_102436            Male  White/Caucasian      2015-11-20     24   
1  IMPACT_105670          Female  White/Caucasian      2015-11-25     24   
2  IMPACT_113135          Female  White/Caucasian      2016-09-12     24   
3  IMPACT_115876            Male       Mixed Race      2016-11-09     24   
4  IMPACT_136775          Female       Mixed Race      2017-11-20     24   

   Wheal (mm)  OUT24NOI  Age at OFC 24  Ara h1 (kU/L)  Ara h2 (kU/L)  \
0         2.5       1.0           79.5           1.14           8.67   
1         0.0       1.0           68.2           1.95          10.40   
2        24.5       0.0           65.3          16.30         181.00   
3        27.5       0.0           58.6           8.75           9.54   
4         7.5       0.0           74.3           0.88           7.80   

   Ara h3 (kU/L)  Ara h6 (kU/L)  Peanut IgE (kU/L)  Total IgE (kU/L)  
0           0.05           6.64               8.41              97.0  
1           0.39           4.97              10.70              70.0  
2           1.40         190.00             243.00             565.0  
3           0.05          11.00              19.20             727.0  
4           0.05           4.93               9.06              25.0

# Renaming columns 

In [82]:
# rename 'OUT24NOI' to 'OFC Pass'
impact_all_24 = impact_all_24.rename(columns={'OUT24NOI': 'OFC Pass'})

# rename Age
impact_all_24 = impact_all_24.rename(columns={'Age at OFC 24': 'Age'})


In [83]:
print(impact_all_24.shape)
impact_all_24.head()

(118, 14)


Participant ID Sex (character)             Race Collection Date  Visit  \
0  IMPACT_102436            Male  White/Caucasian      2015-11-20     24   
1  IMPACT_105670          Female  White/Caucasian      2015-11-25     24   
2  IMPACT_113135          Female  White/Caucasian      2016-09-12     24   
3  IMPACT_115876            Male       Mixed Race      2016-11-09     24   
4  IMPACT_136775          Female       Mixed Race      2017-11-20     24   

   Wheal (mm)  OFC Pass   Age  Ara h1 (kU/L)  Ara h2 (kU/L)  Ara h3 (kU/L)  \
0         2.5       1.0  79.5           1.14           8.67           0.05   
1         0.0       1.0  68.2           1.95          10.40           0.39   
2        24.5       0.0  65.3          16.30         181.00           1.40   
3        27.5       0.0  58.6           8.75           9.54           0.05   
4         7.5       0.0  74.3           0.88           7.80           0.05   

   Ara h6 (kU/L)  Peanut IgE (kU/L)  Total IgE (kU/L)  
0           6.64               8.41              97.0  
1           4.97              10.70              70.0  
2         190.00             243.00             565.0  
3          11.00              19.20             727.0  
4           4.93               9.06              25.0

#### Creating Binary column from Child's Sex -> Male

note origial mapping in the raw eat dataset was:  
'Male', 'Female'  

in this new encoding:  
1=male, 0=female

In [84]:
# function to map the values in 'Sex (character)' to integers
def encode_sex(sex):
    if sex == "Male":
        return 1
    elif sex == "Female":
        return 0
    else:
        return None  # Return None for any other values

# create the new 'Male' column
impact_all_24['Male'] = impact_all_24['Sex (character)'].apply(encode_sex)

# drop original column 
impact_all_24 = impact_all_24.drop(columns=['Sex (character)'])


In [85]:
print(impact_all_24.shape)
impact_all_24.head()

(118, 14)


Participant ID             Race Collection Date  Visit  Wheal (mm)  \
0  IMPACT_102436  White/Caucasian      2015-11-20     24         2.5   
1  IMPACT_105670  White/Caucasian      2015-11-25     24         0.0   
2  IMPACT_113135  White/Caucasian      2016-09-12     24        24.5   
3  IMPACT_115876       Mixed Race      2016-11-09     24        27.5   
4  IMPACT_136775       Mixed Race      2017-11-20     24         7.5   

   OFC Pass   Age  Ara h1 (kU/L)  Ara h2 (kU/L)  Ara h3 (kU/L)  Ara h6 (kU/L)  \
0       1.0  79.5           1.14           8.67           0.05           6.64   
1       1.0  68.2           1.95          10.40           0.39           4.97   
2       0.0  65.3          16.30         181.00           1.40         190.00   
3       0.0  58.6           8.75           9.54           0.05          11.00   
4       0.0  74.3           0.88           7.80           0.05           4.93   

   Peanut IgE (kU/L)  Total IgE (kU/L)  Male  
0               8.41              97.0   1.0  
1              10.70              70.0   0.0  
2             243.00             565.0   0.0  
3              19.20             727.0   1.0  
4               9.06              25.0   0.0

In [86]:
impact_all_24['Race'].unique()

array(['White/Caucasian', 'Mixed Race', 'Asian',
       'Black or African American', nan], dtype=object)

# One Hot Encoding for Race 
note original values for "Race" from the raw IMPACT dataset is as follows
- 'White/Caucasian', 
- 'Mixed Race', 
- 'Asian',
- 'Black or African American'

In [87]:
# Create dummy variables for "race"
race_dummies = pd.get_dummies(impact_all_24['Race'])

# Define the mapping between values and column names
race_mapping = {
    'White/Caucasian': 'White',
    'Black or African American': 'Black',
    'Asian': 'Asian',
    'Mixed Race': 'Mixed',
     np.nan: 'Unknown'
}

# iterate over the mapping and update the dataframe columns
for value, column_name in race_mapping.items():
    if value in race_dummies.columns:
        impact_all_24[column_name] = race_dummies[value].fillna(0).astype(int)
    else:
        impact_all_24[column_name] = 0

# drop original "race" column
impact_all_24 = impact_all_24.drop('Race', axis=1)


In [88]:
print(impact_all_24.shape)
impact_all_24.head()

(118, 18)


Participant ID Collection Date  Visit  Wheal (mm)  OFC Pass   Age  \
0  IMPACT_102436      2015-11-20     24         2.5       1.0  79.5   
1  IMPACT_105670      2015-11-25     24         0.0       1.0  68.2   
2  IMPACT_113135      2016-09-12     24        24.5       0.0  65.3   
3  IMPACT_115876      2016-11-09     24        27.5       0.0  58.6   
4  IMPACT_136775      2017-11-20     24         7.5       0.0  74.3   

   Ara h1 (kU/L)  Ara h2 (kU/L)  Ara h3 (kU/L)  Ara h6 (kU/L)  \
0           1.14           8.67           0.05           6.64   
1           1.95          10.40           0.39           4.97   
2          16.30         181.00           1.40         190.00   
3           8.75           9.54           0.05          11.00   
4           0.88           7.80           0.05           4.93   

   Peanut IgE (kU/L)  Total IgE (kU/L)  Male  White  Black  Asian  Mixed  \
0               8.41              97.0   1.0      1      0      0      0   
1              10.70              70.0   0.0      1      0      0      0   
2             243.00             565.0   0.0      1      0      0      0   
3              19.20             727.0   1.0      0      0      0      1   
4               9.06              25.0   0.0      0      0      0      1   

   Unknown  
0        0  
1        0  
2        0  
3        0  
4        0

In [89]:
# add the missing columns (Flare, h8, h9, Other) and fill with NaNs so the dataset matches LEAP
impact_all_24['Flare (mm)'] = np.nan
impact_all_24['Ara h8 (kU/L)'] = np.nan
impact_all_24['Ara h9 (kU/L)'] = np.nan
impact_all_24['Other'] = np.nan

In [90]:
print(impact_all_24.shape)
impact_all_24.head()

(118, 22)


Participant ID Collection Date  Visit  Wheal (mm)  OFC Pass   Age  \
0  IMPACT_102436      2015-11-20     24         2.5       1.0  79.5   
1  IMPACT_105670      2015-11-25     24         0.0       1.0  68.2   
2  IMPACT_113135      2016-09-12     24        24.5       0.0  65.3   
3  IMPACT_115876      2016-11-09     24        27.5       0.0  58.6   
4  IMPACT_136775      2017-11-20     24         7.5       0.0  74.3   

   Ara h1 (kU/L)  Ara h2 (kU/L)  Ara h3 (kU/L)  Ara h6 (kU/L)  ...  Male  \
0           1.14           8.67           0.05           6.64  ...   1.0   
1           1.95          10.40           0.39           4.97  ...   0.0   
2          16.30         181.00           1.40         190.00  ...   0.0   
3           8.75           9.54           0.05          11.00  ...   1.0   
4           0.88           7.80           0.05           4.93  ...   0.0   

   White  Black  Asian  Mixed  Unknown  Flare (mm)  Ara h8 (kU/L)  \
0      1      0      0      0        0         NaN            NaN   
1      1      0      0      0        0         NaN            NaN   
2      1      0      0      0        0         NaN            NaN   
3      0      0      0      1        0         NaN            NaN   
4      0      0      0      1        0         NaN            NaN   

   Ara h9 (kU/L)  Other  
0            NaN    NaN  
1            NaN    NaN  
2            NaN    NaN  
3            NaN    NaN  
4            NaN    NaN  

[5 rows x 22 columns]

In [91]:
# reorganizing columns 

print(impact_all_24.columns)

Index(['Participant ID', 'Collection Date', 'Visit', 'Wheal (mm)', 'OFC Pass',
       'Age', 'Ara h1 (kU/L)', 'Ara h2 (kU/L)', 'Ara h3 (kU/L)',
       'Ara h6 (kU/L)', 'Peanut IgE (kU/L)', 'Total IgE (kU/L)', 'Male',
       'White', 'Black', 'Asian', 'Mixed', 'Unknown', 'Flare (mm)',
       'Ara h8 (kU/L)', 'Ara h9 (kU/L)', 'Other'],
      dtype='object')


In [92]:
print(impact_all_24.shape)
impact_all_24.head()

(118, 22)


Participant ID Collection Date  Visit  Wheal (mm)  OFC Pass   Age  \
0  IMPACT_102436      2015-11-20     24         2.5       1.0  79.5   
1  IMPACT_105670      2015-11-25     24         0.0       1.0  68.2   
2  IMPACT_113135      2016-09-12     24        24.5       0.0  65.3   
3  IMPACT_115876      2016-11-09     24        27.5       0.0  58.6   
4  IMPACT_136775      2017-11-20     24         7.5       0.0  74.3   

   Ara h1 (kU/L)  Ara h2 (kU/L)  Ara h3 (kU/L)  Ara h6 (kU/L)  ...  Male  \
0           1.14           8.67           0.05           6.64  ...   1.0   
1           1.95          10.40           0.39           4.97  ...   0.0   
2          16.30         181.00           1.40         190.00  ...   0.0   
3           8.75           9.54           0.05          11.00  ...   1.0   
4           0.88           7.80           0.05           4.93  ...   0.0   

   White  Black  Asian  Mixed  Unknown  Flare (mm)  Ara h8 (kU/L)  \
0      1      0      0      0        0         NaN            NaN   
1      1      0      0      0        0         NaN            NaN   
2      1      0      0      0        0         NaN            NaN   
3      0      0      0      1        0         NaN            NaN   
4      0      0      0      1        0         NaN            NaN   

   Ara h9 (kU/L)  Other  
0            NaN    NaN  
1            NaN    NaN  
2            NaN    NaN  
3            NaN    NaN  
4            NaN    NaN  

[5 rows x 22 columns]

In [94]:
# Define the desired column order
desired_order = ['Participant ID',
                 'Collection Date',
                 'Visit',
                 "Age",
                 "Male",
                 "White",
                 "Black",
                 "Asian",
                 "Other",
                 "Mixed",
                 "Unknown",
                 "Wheal (mm)",
                 "Flare (mm)",
                 "Total IgE (kU/L)",
                 "Peanut IgE (kU/L)",
                 "Ara h1 (kU/L)",
                 "Ara h2 (kU/L)",
                 "Ara h3 (kU/L)",
                 "Ara h6 (kUA/L)",
                 "Ara h8 (kU/L)",
                 "Ara h9 (kU/L)",
                 "OFC Pass"]

# Reorder the columns using reindex
impact_all_24 = impact_all_24.reindex(columns=desired_order)


In [95]:
print(impact_all_24.shape)
impact_all_24.head()

(118, 22)


Participant ID Collection Date  Visit   Age  Male  White  Black  Asian  \
0  IMPACT_102436      2015-11-20     24  79.5   1.0      1      0      0   
1  IMPACT_105670      2015-11-25     24  68.2   0.0      1      0      0   
2  IMPACT_113135      2016-09-12     24  65.3   0.0      1      0      0   
3  IMPACT_115876      2016-11-09     24  58.6   1.0      0      0      0   
4  IMPACT_136775      2017-11-20     24  74.3   0.0      0      0      0   

   Other  Mixed  ...  Flare (mm)  Total IgE (kU/L)  Peanut IgE (kU/L)  \
0    NaN      0  ...         NaN              97.0               8.41   
1    NaN      0  ...         NaN              70.0              10.70   
2    NaN      0  ...         NaN             565.0             243.00   
3    NaN      1  ...         NaN             727.0              19.20   
4    NaN      1  ...         NaN              25.0               9.06   

   Ara h1 (kU/L)  Ara h2 (kU/L)  Ara h3 (kU/L)  Ara h6 (kUA/L)  Ara h8 (kU/L)  \
0           1.14           8.67           0.05             NaN            NaN   
1           1.95          10.40           0.39             NaN            NaN   
2          16.30         181.00           1.40             NaN            NaN   
3           8.75           9.54           0.05             NaN            NaN   
4           0.88           7.80           0.05             NaN            NaN   

   Ara h9 (kU/L)  OFC Pass  
0            NaN       1.0  
1            NaN       1.0  
2            NaN       0.0  
3            NaN       0.0  
4            NaN       0.0  

[5 rows x 22 columns]

# Final export for OFC 24 
- manual merger of data collected on different dates (keeping later date)
- removed the 3 participants who did not have OFC results 
    - Only 114 participants made it to week 24 OFC according to IMPACT Chart
    - dataset had 117, and I removed the 3 that did not have OFC results manually:
        - IMPACT_149018
        - IMPACT_695276
        - IMPACT_763786
        

In [97]:
#commenting out to prevent export again 

# #exporting 
# impact_all_24.to_excel('Data/IMPACT_Study/impact_all_24_raw.xlsx', index=False)


# Final import of vOFC 24 clean data for IMPACT study 

In [99]:
impact_all_24_clean = pd.read_excel("Data/IMPACT_Study/impact_all_24_clean.xlsx")


In [100]:
impact_all_24_clean.head()

Participant ID Collection Date  Visit   Age  Male  White  Black  Asian  \
0  IMPACT_102436      2015-11-20     24  79.5     1      1      0      0   
1  IMPACT_105670      2015-11-25     24  68.2     0      1      0      0   
2  IMPACT_113135      2016-09-12     24  65.3     0      1      0      0   
3  IMPACT_115876      2016-11-09     24  58.6     1      0      0      0   
4  IMPACT_136775      2017-11-20     24  74.3     0      0      0      0   

   Other  Mixed  ...  Flare (mm)  Total IgE (kU/L)  Peanut IgE (kU/L)  \
0    NaN      0  ...         NaN                97               8.41   
1    NaN      0  ...         NaN                70              10.70   
2    NaN      0  ...         NaN               565             243.00   
3    NaN      1  ...         NaN               727              19.20   
4    NaN      1  ...         NaN                25               9.06   

   Ara h1 (kU/L)  Ara h2 (kU/L)  Ara h3 (kU/L)  Ara h6 (kUA/L)  Ara h8 (kU/L)  \
0           1.14           8.67           0.05             NaN            NaN   
1           1.95          10.40           0.39             NaN            NaN   
2          16.30         181.00           1.40             NaN            NaN   
3           8.75           9.54           0.05             NaN            NaN   
4           0.88           7.80           0.05             NaN            NaN   

   Ara h9 (kU/L)  OFC Pass  
0            NaN         1  
1            NaN         1  
2            NaN         0  
3            NaN         0  
4            NaN         0  

[5 rows x 22 columns]